woocommerce/woocommerce-ios
# Show different copy in Products tab banner based on Products feature switch in settings

Part 2 (Show different copy in Products tab banner) for #2042 

- [x] ⚠️ Update PR base to `develop` after https://github.com/woocommerce/woocommerce-ios/pull/2275 is merged ⚠️ 

## Changes

- Created a reusable `SwappableSubviewContainerView` as a simple container view that replaces its subview on updates
  - Reorganized the file into a new `Containers` folder in the file hierarchy
<img width="452" alt="Screen Shot 2020-05-14 at 11 02 14 AM" src="https://user-images.githubusercontent.com/1945542/81887890-b6288100-95d2-11ea-8368-405cecd58884.png">
- Created `ProductsTopBannerFactory` that generates a top banner view (`TopBannerView`) based on the app settings
- In `ProductsViewController`, updated the top banner to `SwappableSubviewContainerView` and updated the top banner via `ProductsTopBannerFactory` in the following occasions:
  - `viewDidLoad`
  - when the Products feature switch changes under Settings > Experimental Features via `NotificationCenter` with name `ProductsFeatureSwitchDidChange`

## Testing

- Go to Settings > Experimental Features, and turn off the Products feature switch
- Go to the Products tab --> the WIP banner should have the title and info as in the screenshot below under "Products switch off"
- Go to Settings > Experimental Features, and turn on the Products feature switch
- Go to the Products tab --> the WIP banner should have the title and info as in the screenshot below under "Products switch on"

## Example screencast/screenshots

![simulator](https://user-images.githubusercontent.com/1945542/81888460-f1777f80-95d3-11ea-8bde-9aa94d555fb1.gif)


Products switch off | Products switch on
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-05-14 at 11 10 49](https://user-images.githubusercontent.com/1945542/81888330-b6754c00-95d3-11ea-9fb2-565cacd69bfa.png) | ![Simulator Screen Shot - iPhone 11 - 2020-05-14 at 11 10 58](https://user-images.githubusercontent.com/1945542/81888341-bd9c5a00-95d3-11ea-862d-80c2e3b44c9d.png)


Update release notes:
